pf2-core/Core/Rules/Creature/Alignment.cs

```diff
@@ -1,4 +1,4 @@
-namespace PF2.Core.Rules
+namespace PF2.Core.Rules.Creature
 {
     public enum Alignment
```

Comment:
Just a thought about this class.  For success/failure we have a structure for isSuccess and isFailure.  This seems like a similar situation where we could have isGood or isEvil etc. for the purposes of condition checking for spell resolution.  Not sure which of the boolean / enum structures are better but maybe we want them to at least be consistent?

Reply:
That sounds like a good convention to use. Basically setting up flags for alignment, where if all of them are false, it would put you as true neutral.

Reply:
My only concern is when it comes to spells and actions that have special abilities against creatures of a specific type.  Using an enum is simpler in most cases, except those where we'd have to do alignment = LawfulGood or NeutralGood or ChaoticGood for example.  I think the enum is probably cleaner but that may be me.

Reply:
It might not be easier when it comes to creating immunities and resistances. A spell targeting only Lawful creatures could have code like `if (immunity.Alignment.isLawful()) return Effect();`. that seems to read better than `if (immunity.Alignment == Alignment.Lawful)` (unless there's a more elegant way to write it).

Reply:
In thinking about it, I think that the enum is cleaner when defining the 9 alignments and success/failure, and we could write extension functions like IsLawful(eAlignment alignment) or IsSuccess() to handle immunities and resistances.  when there's a set amount of possible combinations, an enum feels cleaner to me than multiple booleans.

Reply:
Sorry to be late to the conversation, but I think this conversation raises the question of if we're engaging in YAML -- 'Ya ain't gonna need it'.  What is the intended scope of this project?  Character sheets, or a virtual tabletop?  Character Sheets are a much simpler project, a VT much, much more difficult.  We should take that discussion to slack, I think.

Either way, I firmly believe that enum is the right option here.  If you want .isEvil? or .isLawful methods, just define them as an extension on the enum.

https://stackoverflow.com/questions/5985661/methods-inside-enum-in-c-sharp